Source/Controllers/Image Editor/LIFEToolButton.m

```diff
@@ -0,0 +1,116 @@
+//
+//  LIFEToolButton.m
+//  Copyright (C) 2018 Buglife, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//
+
+#import "LIFEToolButton.h"
+#import "LIFEMacros.h"
+
+let kImageHeight = 23.0f;
+
+@interface LIFEToolButton ()
+
+@property (nonnull, nonatomic) NSLayoutConstraint *imageCenterX;
+@property (nonnull, nonatomic) NSLayoutConstraint *imageBottom;
+@property (nonnull, nonatomic) NSLayoutConstraint *imageWidth;
+@property (nonnull, nonatomic) NSLayoutConstraint *imageHeight;
+@property (nonnull, nonatomic) NSMutableDictionary<NSNumber *, UIColor *> *controlStateTintColors;
+
+@end
+
+@implementation LIFEToolButton
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _controlStateTintColors = [[NSMutableDictionary alloc] init];
+        
+        _imageView = [[UIImageView alloc] init];
+        _imageView.contentMode = UIViewContentModeScaleAspectFit;
+        _titleView = [[UILabel alloc] init];
+        _titleView.textAlignment = NSTextAlignmentCenter;
+        _titleView.font = [UIFont systemFontOfSize:10];
+        
+        _imageView.translatesAutoresizingMaskIntoConstraints = NO;
+        _titleView.translatesAutoresizingMaskIntoConstraints = NO;
+        
+        [self addSubview:_imageView];
+        [self addSubview:_titleView];
+        
+//        _imageView.backgroundColor = [[UIColor redColor] colorWithAlphaComponent:0.25];
+//        _titleView.backgroundColor = [[UIColor blueColor] colorWithAlphaComponent:0.25];
+//        self.backgroundColor = [[UIColor greenColor] colorWithAlphaComponent:0.25];
+        
+        CGFloat imageScale = 0.75;
```

Comment:
there seems to be inconsistent usage of f (and .0f) suffixes throughout.

Reply:
`f` is required when using the `let` macro, otherwise it can (and IMHO should) be omitted. But I went ahead and fixed this line.